pkg/generic/thrift/json.go

```diff
@@ -115,8 +151,75 @@ func (m *ReadJSON) SetBinaryWithBase64(enable bool) {
 	m.binaryWithBase64 = enable
 }
 
+// SetDynamicGo ...
+func (m *ReadJSON) SetDynamicGo(t2jBinaryConv, t2jBinaryConvWithException *t2j.BinaryConv, msg remote.Message) {
+	m.msg = msg
+	m.dynamicgoEnabled = true
+	if m.isClient {
+		// set binary conv to handle an exception field
+		m.t2jBinaryConv = *t2jBinaryConvWithException
+	} else {
+		m.t2jBinaryConv = *t2jBinaryConv
+	}
+}
+
 // Read read data from in thrift.TProtocol and convert to json string
 func (m *ReadJSON) Read(ctx context.Context, method string, in thrift.TProtocol) (interface{}, error) {
+	// fallback logic
+	if !m.dynamicgoEnabled {
+		return m.originalRead(ctx, method, in)
+	}
+
+	// dynamicgo logic
+	tProt, ok := in.(*cthrift.BinaryProtocol)
+	if !ok {
+		return nil, perrors.NewProtocolErrorWithMsg("TProtocol should be BinaryProtocol")
+	}
+
+	var tyDsc *dthrift.TypeDescriptor
+	if m.msg.MessageType() == remote.Reply {
+		tyDsc = m.svc.DynamicGoDsc.Functions()[method].Response()
+	} else {
+		tyDsc = m.svc.DynamicGoDsc.Functions()[method].Request()
+	}
+
+	var resp interface{}
+	if tyDsc.Struct().Fields()[0].Type().Type() == dthrift.VOID {
+		if _, err := tProt.ByteBuffer().ReadBinary(voidWholeLen); err != nil {
+			return nil, err
+		}
+		resp = descriptor.Void{}
+	} else {
+		msgBeginLen := bthrift.Binary.MessageBeginLength(method, thrift.TMessageType(m.msg.MessageType()), m.msg.RPCInfo().Invocation().SeqID())
+		transBuff, err := tProt.ByteBuffer().ReadBinary(m.msg.PayloadLen() - msgBeginLen - bthrift.Binary.MessageEndLength())
+		if err != nil {
+			return nil, err
+		}
+
+		// json size is usually 2 times larger than equivalent thrift data
+		buf := mcache.Malloc(len(transBuff) * 2)[0:0]
+		defer mcache.Free(buf)
+		// thrift []byte to json []byte
+		if err := m.t2jBinaryConv.DoInto(ctx, tyDsc, transBuff, &buf); err != nil {
+			return nil, err
+		}
+		if len(buf) > structWrapLen {
+			buf = buf[structWrapLen : len(buf)-1]
```

Comment:
why are we removing the first 4 bytes and the last 1 byte?

Reply:
@felix021 In the original kitex way, it returns json string without the first `{"":` and the last `}`. So we need to remove them.

Reply:
Since it's a bit weird, I suggest we add a function `removePrefixAndSuffix(buf)` here and add some comments.

Btw, are you sure the prefix is `{"":`, an empty json key? Doesn't make sense at all...

Reply:
@felix021 Ok, I'll add a function.
And I'm very sure that the prefix is always `{"":` because dynamicgo writes `{` first and `"response_field_alias":` second, and the alias name of response is always empty.